include/deal.II/lac/arpack_solver.h

```diff
@@ -139,13 +157,20 @@ class ArpackSolver : public Subscriptor
 
   /**
    * Solve the generalized eigensprectrum problem $A x=\lambda B x$ by calling
-   * the <code>dneupd</code> and <code>dnaupd</code> functions of ARPACK.
+   * the <code>dsaupd</code> and <code>dseupd</code> or
+   * <code>dnaupd</code> and <code>dneupd</code> functions of ARPACK.
    *
    * The function returns a vector of eigenvalues of length <i>n</i> and a
-   * vector of eigenvectors, where the latter should be twice the size of the
-   * eigenvalue vector. The first <i>n</i> vectors in
-   * <code>eigenvectors</code> will be the real parts of the eigenvectors, the
-   * second <i>n</i> the imaginary parts.
+   * vector of eigenvectors of length <i>n</i> in the symmetric case
+   * and of length <i>n+1<i> in the non-symmetric case. In the symmetric case
```

Comment:
are you sure it's `n+1` instead of `n%2+1`? Say the user needs 3 eigenpais in non-symmetric case, then my understanding of your description is that Arpack will be fine with it. However, if the user needs 4, then Arpack requires space to store 5, right?